### PR TITLE
rbd: add functions to fetch mirroring status of a RBD image

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Added two functions:
1. GetMirrorImageInfo()
2. GetImageMirrorMode()

Signed-off-by: Mudit Agarwal <muagarwa@redhat.com>

Fixes: #379 

## Checklist
- [ ] Added tests for features and functional changes
- [ ] Public functions and types are documented
- [ ] Standard formatting is applied to Go code

### CODE REVIEW

File: rbd/mirror.go [hunk 1]
```diff
@@ -165,3 +165,66 @@ func (image *Image) MirrorInstanceID() (string, error) {
...
+//                                rbd_mirror_image_info_t *mirror_image_info,
+//                                size_t info_size)
+func (image *Image) GetMirrorImageInfo() (*MirrorImageInfo, error) {
+	if err := image.validate(imageIsOpen); err != nil {
+		return nil, err
+	}
+
+	var cInfo C.rbd_mirror_image_info_t
+
+	ret := C.rbd_mirror_image_get_info(
```
Comment (phlogistonjohn) [inline rbd/mirror.go:204]: I'm pretty sure this function allocates memory and thus we need a call to rbd_mirror_image_get_info_cleanup to free the memory after we've converted it to go types.

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Perhaps I am overlooking it but I think this needs a call to rbd_mirror_image_get_info_cleanup, because rbd_mirror_image_get_info allocates memory.

File: rbd/mirror.go (hunk 1 above)
Comment (agarwal-mudit) [reply rbd/mirror.go:204]: added

**Review by phlogistonjohn — APPROVED** [review]
Looks good, thanks!

**Review by ansiwen — APPROVED** [review]
LGTM

**Pull request merged**